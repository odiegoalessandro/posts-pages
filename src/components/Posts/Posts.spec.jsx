import { render, screen } from "@testing-library/react";
import { Posts } from ".";

const postsMock = { 
  posts: [
    {
      id: 1,
      title: 'title 1',
      body: 'body',
      cover: 'img/img1.png'
    },
    {
      id: 2,
      title: 'title 2',
      body: 'body',
      cover: 'img/img2.png'
    },
    {
      id: 3,
      title: 'title 3',
      body: 'body',
      cover: 'img/img3.png'
    }
  ]
};
describe('<Posts />', () => {
  it('should render Posts', () => { 
    render(<Posts {...postsMock} />);
    
    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole('img')).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);

    const lastImage = screen.getByRole('img', { name: /title 3/i });

    expect(lastImage).toHaveAttribute('src', 'img/img3.png');
  });

  it('should not render Posts', () => {
    render(<Posts />);
    
    expect(screen.queryByRole('heading', { name: /title/i })).not.toBeInTheDocument();
  });

  it('should match to snapshot', () => {
    const { asFragment } = render(<Posts {...postsMock} />);

    expect(asFragment()).toMatchSnapshot();
  });
});