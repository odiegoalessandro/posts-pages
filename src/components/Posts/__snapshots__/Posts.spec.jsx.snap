// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match to snapshot 1`] = `
<DocumentFragment>
  <div
    class="post-container"
  >
    <div
      class="post"
    >
      <img
        alt="title 1"
        src="img/img1.png"
      />
      <div
        class="post-content"
      >
        <h2>
          title 1
        </h2>
        <p>
          body
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="title 2"
        src="img/img2.png"
      />
      <div
        class="post-content"
      >
        <h2>
          title 2
        </h2>
        <p>
          body
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="title 3"
        src="img/img3.png"
      />
      <div
        class="post-content"
      >
        <h2>
          title 3
        </h2>
        <p>
          body
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
